src/tpg_test_mgmt_cli.c

```diff
@@ -186,6 +186,78 @@ cmdline_parse_inst_t cmd_tests_stop = {
         NULL,
     },
 };
+/****************************************************************************
+ * - "show link rate"
+ ****************************************************************************/
+struct cmd_show_link_rate_result {
+    cmdline_fixed_string_t show;
+    cmdline_fixed_string_t link;
+    cmdline_fixed_string_t rate;
+};
+
+static cmdline_parse_token_string_t cmd_show_link_rate_T_show =
+TOKEN_STRING_INITIALIZER(struct cmd_show_link_rate_result, show, "show");
+static cmdline_parse_token_string_t cmd_show_link_rate_T_link =
+TOKEN_STRING_INITIALIZER(struct cmd_show_link_rate_result, link, "link");
+static cmdline_parse_token_string_t cmd_show_link_rate_T_rate =
+TOKEN_STRING_INITIALIZER(struct cmd_show_link_rate_result, rate, "rate");
+
+static void tpg_print_line_rate(void)
```

Comment:
This code will now be duplicated in tpg_test_mgmt_cli.c and tpg_test_stats.c. Isn't it better to add a function that displays the link stats on a generic printer. For example, the per test statistics are dumped with:

```
void test_state_show_stats(const tpg_test_case_t *te,
                           printer_arg_t *printer_arg);
```

Why not define a function for printing link rate in a generic way and call that one both from the CLI (in `cmd_show_link_rate_parsed`) and from the UI (in `test_display_stats_hdr`)? A prototype could be:

```
void test_show_link_rate(uint32_t port, printer_arg_t *printer_arg);
```